SMU.cpp

```diff
@@ -246,6 +244,7 @@ void SessionItem::downloadFromUrl(QString url)
     QUrl firmwareUrl(url);
     m_firmware_fd = new FileDownloader(firmwareUrl, this);
     connect(m_firmware_fd, SIGNAL (downloaded()), this, SLOT (handleDownloadedFirmware()));
+    delete m_firmware_fd;
```

Comment:
Why delete it? handleDownloadedFirmware() needs the data contained by m_firmware_fd as soon as the downloaded() signal is emitted.